backend/src/main/java/org/eclipse/tractusx/puris/backend/production/logic/adapter/PlannedProductionSammMapper.java

```diff
@@ -0,0 +1,145 @@
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.eclipse.tractusx.puris.backend.production.logic.adapter;
+
+import org.eclipse.tractusx.puris.backend.common.domain.model.measurement.ItemQuantityEntity;
+import org.eclipse.tractusx.puris.backend.masterdata.domain.model.Material;
+import org.eclipse.tractusx.puris.backend.masterdata.domain.model.Partner;
+import org.eclipse.tractusx.puris.backend.masterdata.logic.service.MaterialPartnerRelationService;
+import org.eclipse.tractusx.puris.backend.production.domain.model.OwnProduction;
+import org.eclipse.tractusx.puris.backend.production.domain.model.ReportedProduction;
+import org.eclipse.tractusx.puris.backend.production.logic.dto.plannedproductionsamm.*;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import lombok.extern.slf4j.Slf4j;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@Slf4j
+public class PlannedProductionSammMapper {
+    @Autowired
+    private MaterialPartnerRelationService mprService;
+
+    public PlannedProductionOutput ownProductionToSamm(List<OwnProduction> production) {
+        if (production == null || production.isEmpty()) {
+            log.warn("Can't map empty list");
+            return null;
+        }
+        Partner partner = production.get(0).getPartner();
+        Material material = production.get(0).getMaterial();
+        if (production.stream().anyMatch(prod -> !prod.getPartner().equals(partner))) {
+            log.warn("Can't map production list with different partners");
+            return null;
+        }
+
+        if (production.stream().anyMatch(prod -> !prod.getMaterial().equals(material))) {
+            log.warn("Can't map production list with different materials");
+            return null;
+        }
+        var groupedByPositionAttributes = production
+                .stream()
+                .collect(Collectors.groupingBy(prod -> new PositionsMappingHelper(
+                    prod.getCustomerOrderNumber(),
+                    prod.getSupplierOrderNumber(),
+                    prod.getCustomerOrderPositionNumber()
+                )));
+        PlannedProductionOutput samm = new PlannedProductionOutput();
+
+        samm.setMaterialGlobalAssetId(material.getMaterialNumberCx());
+        samm.setMaterialNumberCustomer(material.getOwnMaterialNumber());
+        samm.setMaterialNumberSupplier(material.getOwnMaterialNumber());
```

Comment:
There is no more any materialNumberCustomer or materialNumberSupplier in PlannedProductionOutput:2.0.0

Oh dear, seems like I missed that in the recent review. You can see the changelog [here](https://github.com/eclipse-tractusx/sldt-semantic-models/blob/main/io.catenax.planned_production_output/RELEASE_NOTES.md).

In short:
- collections are now sets
- timestamps are a default characteristic
- only the global asset id is used to identify materials (now mandatory)

Please update also the samm accordingly